app/src/main/java/com/islamey/hamza/wazaker/ui/EveningAzkarFragment/EveningAzkarAdapter.kt

```diff
@@ -0,0 +1,74 @@
+package com.islamey.hamza.wazaker.ui.EveningAzkarFragment
+
+
+import android.annotation.SuppressLint
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.AsyncListDiffer
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import com.islamey.hamza.wazaker.domain.Models.ZekerModel
+import com.islamey.hamza.wazaker.utils.Utils.shareText
+import com.islamey.wazkar.databinding.MorningItemBinding
+
+class EveningAzkarAdapter : RecyclerView.Adapter<EveningAzkarAdapter.ViewHolder>() {
+
+    inner class ViewHolder(val binding: MorningItemBinding) :
+        RecyclerView.ViewHolder(binding.root)
+
+    private val diffCallback = object : DiffUtil.ItemCallback<ZekerModel>() {
+        override fun areItemsTheSame(oldItem: ZekerModel, newItem: ZekerModel): Boolean {
+            return oldItem.zeker == newItem.zeker
+        }
+
+        override fun areContentsTheSame(oldItem: ZekerModel, newItem: ZekerModel): Boolean {
+            return oldItem == newItem
+        }
+    }
+
+    val differ = AsyncListDiffer(this, diffCallback)
+
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int
+    ): EveningAzkarAdapter.ViewHolder {
+        val view = MorningItemBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return ViewHolder(view)
+    }
+
+    @SuppressLint("SetTextI18n")
+    override fun onBindViewHolder(holder: EveningAzkarAdapter.ViewHolder, position: Int) {
+
+        val currentItem = differ.currentList[position]
+        var count = 0
+
+        holder.binding.zekertext.text = currentItem.zeker
+        holder.binding.copybtn.setOnClickListener {
+            shareText(holder.itemView.context, currentItem.zeker)
+        }
+
+
+        holder.binding.plustext.text = count.toString()
+
+        holder.binding.plusButton.setOnClickListener {
+            count++
+            holder.binding.plustext.text = count.toString()
+            if (count >= currentItem.counter) {
+                removeItem(currentItem)
+            }
+        }
```

Comment:
binding actions should be inside viewholder 

Nice to have: make actions outside the adapter